src/scan.ts

```diff
@@ -116,50 +123,115 @@ export async function getActiveContracts(
-                                          choices[0].can_choose_between,
-                                      invalidBefore: timeBefore5Minutes,
-                                      invalidHereafter: timeAfter5Minutes,
-                                  });
-                        }
-                    )
-                )
-            )
+    const addressResolvable: OracleRequest[] = [];
+    const charli3ResolvableData: [CanChoose, ContractHeader][] = [];
+    const charli3Resolvable: OracleRequest[] = [];
+
+    for (const contract of allContractHeaders) {
+        const nextSteps = await client.getNextStepsForContract(
+            contract.contractId,
+            mkEnvironment(timeBefore5Minutes)(timeAfter5Minutes),
+            []
+        )();
+
+        match(
+            (_) => left('Error on next query'),
+            (value: Next) => {
+                value.applicable_inputs.choices.forEach((choice) => {
+                    if (
+                        methods.address &&
+                        isResolvable(
+                            choice,
+                            methods.address.mosAddress,
+                            methods.address.choiceNames
+                        )
+                    ) {
+                        const newRequest: OracleRequest = {
+                            contractId: contract.contractId,
+                            choiceId: choice.for_choice,
+                            choiceBounds: choice.can_choose_between,
+                            invalidBefore: timeBefore5Minutes,
+                            invalidHereafter: timeAfter5Minutes,
+                            bridgeUtxo: none,
+                        };
+
+                        addressResolvable.push(newRequest);
+                    } else if (
+                        methods.charli3 &&
+                        isResolvable(
+                            choice,
+                            { role_token: methods.charli3.roleNames },
+                            [methods.charli3.choiceNames]
+                        )
+                    ) {
+                        charli3ResolvableData.push([choice, contract]);
+                    }
+                });
+            }
+        )(nextSteps);
+    }
+
+    if (methods.charli3 && charli3ResolvableData.length > 0) {
+        const bridgeUtxos = await lucid.utxosAt(methods.charli3.bridgeAddress);
+
+        for (const [choice, contract] of charli3ResolvableData) {
+            const roleMintingPolicy = unPolicyId(
+                contract.roleTokenMintingPolicyId
+            );
+            if (roleMintingPolicy === '') continue;
+
+            const assetClass =
+                roleMintingPolicy +
+                Buffer.from(methods.charli3.roleNames, 'utf-8').toString('hex');
+
+            const utxo = bridgeUtxos.find(
+                (utxo) => utxo.assets[assetClass] === 1n
+            );
+
+            if (!utxo) continue;
```

Comment:
```suggestion
            if (!utxo) {
                scanLogger.debug("Skipping contract: ", contract.contractId, ". Reason: No Bridge UTxO found")
                continue
            };
```

Maybe this pattern for this and the previous continue can be good. To help debugging if a particular contract is not being detected as resolvable.